be/src/formats/parquet/metadata.cpp

```diff
@@ -447,15 +447,15 @@ StatusOr<FileMetaDataPtr> FileMetaDataParser::get_file_metadata() {
         return file_metadata_ptr;
     }
 
-    DataCacheHandle cache_handle;
+    ObjectCacheHandle* cache_handle = nullptr;
     std::string metacache_key =
             _build_metacache_key(_file->filename(), _datacache_options->modification_time, _file_size);
     {
         SCOPED_RAW_TIMER(&_scanner_ctx->stats->footer_cache_read_ns);
-        Status st = _cache->read_object(metacache_key, &cache_handle);
+        Status st = _cache->lookup(metacache_key, &cache_handle);
         if (st.ok()) {
             _scanner_ctx->stats->footer_cache_read_count += 1;
-            return *(static_cast<const FileMetaDataPtr*>(cache_handle.ptr()));
+            return *(static_cast<const FileMetaDataPtr*>(_cache->value(cache_handle)));
```

Comment:
Should call `_cache->relase(cache_handle)` after the value is acquired, otherwise the cache item will be hold by the handle and never be evicted by the cache library.